### PR TITLE
allow kpt setters to set on kustomize inline patches

### PR DESCRIPTION
**Pull request by natasha41575**
Close https://github.com/GoogleContainerTools/kpt/issues/1342

Request from users to allow kpt setters to work on kustomize patches. To do so the inline patches need additional parsing. 

Original request in comment here: https://github.com/GoogleContainerTools/kpt/issues/1218#issuecomment-756486462

### CODE REVIEW

File: kyaml/setters2/add_test.go [hunk 1]
```diff
@@ -43,6 +43,41 @@ metadata:
...
+      path: placeholder
+      value: known
+ `,
+			expected: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+patches:
+- target:
+    kind: Something
+  patch: |
```
Comment (phanimarupaka) [inline kyaml/setters2/add_test.go:74]: Add a case with |-

File: kyaml/setters2/add_test.go [hunk 2]
```diff
@@ -43,6 +43,41 @@ metadata:
...
+    - op: add
+      path: placeholder # {"$openapi":"path"}
```
Comment (phanimarupaka) [inline kyaml/setters2/add_test.go:76]: Have you tested set operation on this ?

File: kyaml/setters2/walk.go [hunk 3]
```diff
@@ -70,11 +72,44 @@ func acceptImpl(v visitor, object *yaml.RNode, p string, oa *openapi.ResourceSch
...
+				values[mapContent[0]] = mapContent[1]
+			}
+		}
+		rn := yaml.NewMapListRNode(&values)
+		if rn != nil {
+			err := acceptImpl(v, rn, p, oa, settersSchema)
+			if err != nil { // do nothing
+				return
+			}
+			str, _ := rn.String()
```
Comment (phanimarupaka) [inline kyaml/setters2/walk.go:104]: You should not ignore the error.

File: kyaml/setters2/walk.go [hunk 4]
```diff
@@ -70,11 +72,44 @@ func acceptImpl(v visitor, object *yaml.RNode, p string, oa *openapi.ResourceSch
 	case yaml.ScalarNode:
 		// Visit the scalar field
 		fieldSchema := getSchema(object, oa, "", settersSchema)
+		if strings.Contains(p, "patch") {
+			parseInlinePatch(v, object, p, oa, settersSchema)
```
Comment (phanimarupaka) [inline kyaml/setters2/walk.go:76]: I think we should do it for any inline strings if we are going down this route. Doing it just for kustomize patches looks hacky.

**Comment by natasha41575** [comment]
closing this PR for now as it needs broader discussion

**Comment by yuwenma** [comment]
@natasha41575 Since @anna is picking up this issue, can you share more context about the previous offline discussion you have with @phanimarupaka , and also your thinkings related to the "broader discussion"?

**Comment by natasha41575** [comment]
@yuwenma Unfortunately I don't remember the conversation :(

From the above comments, seems like my solution here was too hacky and the code needs to be more general than what I did; I specifically targeted kustomize patches, but we need to apply the fix for any inline string.

**Pull request closed without merging**